packages/language/src/pli.langium

```diff
@@ -0,0 +1,666 @@
+    (
+        options+=Options |
+        variable+='VARIABLE' |
+        limited+='LIMITED' |
+        returns+=ReturnsOption |
+        (('EXTERNAL' | 'EXT') ('(' environmentName+=Expression ')')?)
+    )*
+    
+;
+
+ReturnsOption: 'RETURNS' '(' returnAttribute=DeclarationAttribute* ')';
+
+EntryDescription: EntryParameterDescription | EntryUnionDescription;
+
+EntryParameterDescription: (attributes+=DeclarationAttribute+ | star?='*' attributes+=DeclarationAttribute*);
+
+EntryUnionDescription: init=NUMBER attributes+=DeclarationAttribute* ',' prefixedAttributes+=PrefixedAttribute*;
+
+PrefixedAttribute: level=NUMBER attribute=DeclarationAttribute*;
+
+ProcedureParameter: id=FeatureID;
+
+ReferenceItem: ref=[NamedElement:FeatureID] dimensions=Dimensions?;
+
+Expression: BitOrExpression;
+
+
+// Note for expressions: Documentation uses '¬' to denote inversion or negation. The language actually uses '^'
+
+// Priority 7
+BitOrExpression infers Expression: BitAndExpression ({infer BitOrExpression.left=current} op=('|' | '¬' | '^') right=BitAndExpression)*;
+
+// Priority 6
+BitAndExpression infers Expression: CompExpression ({infer BitAndExpression.left=current} op='&' right=CompExpression)*;
+
+// Priority 5
+CompExpression infers Expression: ConcatExpression ({infer CompExpression.left=current} op=('<' | '¬<' | '<=' | '=' | '¬=' | '^=' | '<>' | '>=' | '>' | '¬>') right=ConcatExpression)*;
+
+// Priority 4
+ConcatExpression infers Expression: AddExpression ({infer ConcatExpression.left=current} op=('||' | '!!') right=AddExpression)*;
+
+// Priority 3
+AddExpression infers Expression: MultExpression ({infer AddExpression.left=current} op=('+' | '-' ) right=MultExpression)*;
+
+// Priority 2
+MultExpression infers Expression: ExpExpression ({infer MultExpression.left=current} op=('*' | '/' ) right=ExpExpression)*;
+
+// Priority 1
+ExpExpression infers Expression: PrimaryExpression ({infer ExpExpression.left=current} op='**' right=PrimaryExpression)*;
+
+PrimaryExpression infers Expression: 
+    Literal
+    | '(' Expression ')'
+    | UnaryExpression
+    | LocatorCall;
+
+MemberCall:
+    element=ReferenceItem
+	({infer MemberCall.previous=current} 
+        "." element=ReferenceItem
+    )*;
+
+LocatorCall:
+    element=MemberCall
+    ({infer LocatorCall.previous=current}
+        (pointer?="->" | handle?="=>") element=MemberCall
+    )*;
+
+ProcedureCall: procedure=[ProcedureStatement:ID]  ('(' (args+=(Expression | '*') (',' args+=(Expression | '*'))*)? ')')?;
+
+LabelReference: label=[LabelPrefix:ID];
+
+UnaryExpression: op=('+' | '-' | '¬' | '^') expr=Expression;
+
+ConstantExpression: Literal;
+
+Literal: ('(' multiplier=NUMBER ')')? value=(StringLiteral | NumberLiteral);
+
+StringLiteral: value=STRING_TERM;
+NumberLiteral: value=NUMBER;
+
+FQN returns string: ID ('.' ID)*;
+FeatureID returns string: ID;
+
+hidden terminal WS: /\s+/;
+terminal ExecFragment: /(?<=EXEC\s*)[a-zA-Z]+\s[^;]*/i;
+terminal ID: /[$@#_a-zA-Z][\w_$@#]*/;
```

Comment:
Extralingual characters are allowed in identifiers via the usage of [NAMES](https://www.ibm.com/docs/en/SSY2V3_6.1/pdf/pg.pdf#page=113) compiler option. The default is `@#$` but other characters are allowed as well.

Reply:
Created an issue for this https://github.com/zowe/zowe-pli-language-support/issues/30